internal/controllers/device_data_controller_test.go

```diff
@@ -727,3 +728,90 @@ func TestUserDevicesController_GetUserDevicesErrorCodeQueries(t *testing.T) {
+	app := fiber.New()
+	app.Post("/user/devices/:userDeviceID/error-codes/clear", test.AuthInjectorTestHandler(testUserID), c.ClearUserDeviceErrorCodeQuery)
+
+	t.Run("POST - clear last saved error code response for current user devices", func(t *testing.T) {
+		autoPiInteg := test.BuildIntegrationGRPC(constants.AutoPiVendor, 10, 0)
+		dd := test.BuildDeviceDefinitionGRPC(ksuid.New().String(), "Toyota", "Camry", 2023, autoPiInteg)
+		ud := test.SetupCreateUserDevice(t, testUserID, dd[0].DeviceDefinitionId, nil, "", pdb)
+
+		testData := []struct {
+			Codes      []string
+			OpenAIResp []services.ErrorCodesResponse
+		}{
+			{
+				Codes: []string{"P0017"},
+				OpenAIResp: []services.ErrorCodesResponse{
+					{
+						Code:        "P0017",
+						Description: "Engine Coolant Temperature Circuit Malfunction: This code indicates that the engine coolant temperature sensor is sending a signal that is outside of the expected range, which may cause the engine to run poorly or overheat.",
+					},
+				},
+			},
+			{
+				Codes: []string{"P0016"},
+				OpenAIResp: []services.ErrorCodesResponse{
+					{
+						Code:        "P0016",
+						Description: "Engine Coolant Temperature Circuit Malfunction: This code indicates that the engine coolant temperature sensor is sending a signal that is outside of the expected range, which may cause the engine to run poorly or overheat.",
+					},
+				},
+			},
+		}
+
+		for _, tData := range testData {
+			chtJSON, err := json.Marshal(tData.OpenAIResp)
+			assert.NoError(t, err)
+
+			currTime := time.Now().UTC().Truncate(time.Microsecond)
+			erCodeQuery := models.ErrorCodeQuery{
+				ID:                 ksuid.New().String(),
+				UserDeviceID:       ud.ID,
+				ErrorCodes:         tData.Codes,
+				CodesQueryResponse: null.JSONFrom(chtJSON),
+				CreatedAt:          currTime,
+			}
+
+			err = erCodeQuery.Insert(ctx, pdb.DBS().Writer, boil.Infer())
+			assert.NoError(t, err)
+		}
+
+		request := test.BuildRequest("POST", fmt.Sprintf("/user/devices/%s/error-codes/clear", ud.ID), "")
+		response, _ := app.Test(request)
+		body, _ := io.ReadAll(response.Body)
+
+		assert.Equal(t, fiber.StatusOK, response.StatusCode)
+
+		errCodeQuery, err := models.ErrorCodeQueries(
+			models.ErrorCodeQueryWhere.ClearedAt.IsNotNull(),
+			qm.OrderBy(models.ErrorCodeQueryColumns.CreatedAt+" DESC"),
+			qm.Limit(1),
+		).One(ctx, pdb.DBS().Reader)
+		assert.NoError(t, err)
+
+		//  currTime.Format(time.RFC3339Nano)
+		assert.JSONEq(t,
+			fmt.Sprintf(`{"errorCodes":%s, "clearedAt":"%v"}`, string(errCodeQuery.CodesQueryResponse.JSON), errCodeQuery.ClearedAt.Time),
```

Comment:
There is, unfortunately, a difference between the standard Go string and the RFC-3339 format they use for JSON. Easy fix.